src/coreclr/tests/src/profiler/unittest/getappdomainstaticaddress.cs

```diff
@@ -48,6 +48,8 @@ private static void LoadCollectibleAssembly()
             object instance = Activator.CreateInstance(testType);
 
             Console.WriteLine(instance.GetHashCode());
+
+            collectibleContext.Unload();
```

Comment:
@davmason I am not sure I understand the purpose of this change. The commit title says "prevent ALC from being finalized in test". While this change requests unload and disables finalization, the unload will still happen at some point in the future when all references to the collectibleContext, the assemblies loaded into it and all instances of types from those assemblies are gone. So there is not much difference from what finalizer would do.

Reply:
Before this change the ALC was being intermittently collected and finalized before the call to GetType, causing failures in the test. By adding the call to Unload at the end of the method there is something rooting it and preventing it from being collected and finalized prematurely. So this isn’t about disabling finalization altogether, but preventing it from happening before the test is done using the assemblies loaded in that context.

Reply:
This is a mere chance that its life span is extended this way. Why don't you just keep the reference to the collectibleContext e.g. in a static field if you want it to not to be collected over the period of the test?

Reply:
I do want it to be collected in the test, I am testing profiler behavior with unloadable statics. The behavior that I want is that the test loads an assembly and unloads it, then verify that GetAppDomainStaticAddress always either returns a valid object or CORPROF_E_DATAINCOMPLETE. 

The problem is that before this change the ALC wouldn't last for the whole LoadCollectibleAssembly method. It would work sometimes but if a GC happened at the wrong point the call to `dynamicLibrary.GetType` would crash the runtime because the ALC was collected and then finalized.

Having the call to unload at the end of the method forces the behavior I want, where the assembly is loaded and then unloaded.

Reply:
Ok, got it, then your change would work. You can also use GC.KeepAlive(collectibleContext) instead of the collectibleContext.Unload(); That would make it clearer that the purpose is to keep the collectibleContext alive.